biz.aQute.bndlib/src/aQute/bnd/component/DSAnnotationReader.java

```diff
@@ -633,7 +633,9 @@ private void processActivationObject(String propertyDefKey, ClassTypeSignature p
 			for (Entry<String, Object> entry : componentPropertyAnnotation.entrySet()) {
 				String key = entry.getKey();
 				Object value = entry.getValue();
-				handleValue(key, value, value instanceof TypeRef, null);
+				boolean isClass = value instanceof TypeRef || (value.getClass()
+					.isArray() && ((Object[]) value)[0] instanceof TypeRef);
```

Comment:
I think you also need to handle the case where the array is empty. e.g. `@ClassArrayProp({})`. So the `isClass` test should be like:

```
boolean isClass = value instanceof TypeRef || (value.getClass()
	.isArray() && (Array.getLength(value) > 0) && (Array.get(value, 0) instanceof TypeRef));
```

Perhaps another test with an empty array can be added too.